price-estimator/src/infallible_price_source.rs

```diff
@@ -2,29 +2,26 @@ use anyhow::Result;
 use core::{
     models::TokenId,
     price_estimation::{average_price_source, price_source::PriceSource},
-    token_info::TokenBaseInfo,
+    token_info::{TokenBaseInfo, TokenInfoFetching},
 };
 use pricegraph::Pricegraph;
-use std::{collections::HashMap, num::NonZeroU128};
+use std::{collections::HashMap, num::NonZeroU128, sync::Arc};
 use tokio::sync::{RwLock, RwLockReadGuard};
 
 /// Roughly like `PriceSource` but is updated externally and cannot fail.
 #[derive(Debug, Default)]
 pub struct InfalliblePriceSource {
-    token_infos: HashMap<TokenId, TokenBaseInfo>,
+    tokens: HashMap<TokenId, TokenBaseInfo>,
```

Comment:
no need to change, but I think a somewhat non-obvious design aspect about this component is that it allows immediate access to this information whereas the other PriceSources have an async interface (arguably the interface makes this "obvious" but the name doesn't). 
Therefore, maybe calling this component `InfalliblePrice**Cache**` and the other component `InfalliblePriceCacheUpdater` could be slightly more expressive.

Reply:
I also agree the name makes it seem like a `PriceSource` implementation that never fails. Maybe it should just be called `PriceCache` (and `PriceCacheUpdater`)?